packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -229,7 +229,11 @@ const ProductList = (props) => {
             }
         }
 
-        navigate(`/category/${params.categoryId}?${stringifySearchParams(searchParamsCopy)}`)
+        if (params.categoryId) {
```

Comment:
Thanks, @yunakim714 for fixing this old bug with a simple solution!

**Optional nit**: Above we use a similar conditional to determine when we are on a search results page or a Product List Page (PLP).
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/0d720e6b9bd3b528f5abbaa1b7c090c04d0f9993/packages/template-retail-react-app/app/pages/product-list/index.jsx#L109-L111

The `searchQuery` props is defined inside `ProductList.getProps` and then sent to `<ProductList />` component.
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/0d720e6b9bd3b528f5abbaa1b7c090c04d0f9993/packages/template-retail-react-app/app/pages/product-list/index.jsx#L585 

Ideally, we are consistent and use the same condition on all our conditionals where we branch and implement specific behaviours for the search results page and the PLP. 
That way if the result of the condition changes, we are sure the result presented to the user changes accordingly.

**TL;DR**
In my opinion, we are a bit more consistent if we use `!searchQuery` here:
```suggestion
        if (!searchQuery) {
```
Either way works great, it's an opinion, so I leave it up to you to make a decision.

Reply:
Thanks for the comments @adamraya ! Decided to go with the more consistent solution :)